aws/resource_aws_appsync_api_key.go

```diff
@@ -0,0 +1,166 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appsync"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsAppsyncApiKey() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceAwsAppsyncApiKeyCreate,
+		Read:   resourceAwsAppsyncApiKeyRead,
+		Update: resourceAwsAppsyncApiKeyUpdate,
+		Delete: resourceAwsAppsyncApiKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Managed by Terraform",
+			},
+			"appsync_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"valid_till_date": {
```

Comment:
A few notes here:
* We should match the AWS API and create a single attribute `expires` rather than provide multiple conflicting attributes for the same information (drop `valid_till_date`, `validity_period_days`, and `expiry_date`)
* It should accept a RFC3339 timestamp for now and can be validated via `ValidateFunc: validation.ValidateRFC3339TimeString`